blueocean-dashboard/src/main/js/creation/bitbucket/server/steps/BbChooseServerStep.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { PropTypes } from 'react';
+import { observer } from 'mobx-react';
+import { Dropdown, FormElement } from '@jenkins-cd/design-language';
+
+import FlowStep from '../../../flow2/FlowStep';
+import BbAddServerDialog from '../BbAddServerDialog';
+
+
+let t = null;
+
+
+@observer
+class BbChooseServerStep extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedServer: null,
+            showAddServerDialog: false,
+            urlErrorMsg: null,
+        };
+
+        this.dropdown = null;
+    }
+
+    componentWillMount() {
+        t = this.props.flowManager.translate;
+    }
+
+    _onChangeServerDropdown(option) {
+        const { serverManager } = this.props.flowManager;
+        serverManager.validateVersion(option.id)
+            .then(
+                success => this._onValidateVersion(success),
+                error => this._onValidateVersion(error),
```

Comment:
Since `BbServerManager` explicitly handles the error from `BbServerApi` and transforms it to a result, I don't think you need the second handler here, just one should do it.